test/tools/jsontests/BlockChainTests.cpp

```diff
@@ -628,9 +644,21 @@ void testBCTest(json_spirit::mObject const& _o)
 
     State postState(State::Null); //Compare post states
     BOOST_REQUIRE((_o.count("postState") > 0));
-    ImportTest::importState(_o.at("postState").get_obj(), postState);
-    ImportTest::compareStates(postState, testChain.topBlock().state());
-    ImportTest::compareStates(postState, blockchain.topBlock().state());
+    if (_o.at("postState").type() == json_spirit::Value_type::obj_type)
+    {
+        ImportTest::importState(_o.at("postState").get_obj(), postState);
+        ImportTest::compareStates(postState, testChain.topBlock().state());
+        ImportTest::compareStates(postState, blockchain.topBlock().state());
+    }
+    else
+    {
+        // Attempt to read hash instead of the full state
+        BOOST_REQUIRE(_o.at("postState").type() == json_spirit::Value_type::str_type);
+        string postHash = _o.at("postState").get_str();
+        string message = testName + "Final StateRoot is different! (Look at expect section in the test /src Filler file, or run the test generation with --fullstate flag to geth the full post state.";
+        BOOST_CHECK_MESSAGE(postHash == toHexPrefixed(testChain.topBlock().state().rootHash()), message);
```

Comment:
Probably better the other way around `h256{postHash} == testChain.topBlock().state().rootHash()`
(and do conversion to `h256` only once for these two checks)